### ISSUE TITLE
additive and multiplicative blending

### ISSUE THREAD

**Issue by bil-bas** [Mon-Ouie/ray#44]
I have used these alpha blending options to great effect in the past, in Wrath and Alpha Channel, and miss them from Ray (well, strictly I have only used additive for "simple" lighting , but multiplicative would work well for shadows or other effects). I thought I'd be able to do this with shaders, but shading is apparently done quite separately to blending in the pipeline. Of course, one could argue that I add lighting to the shader in every texture being drawn in a similar way to that used in 3D, but that seems to be overkill to me (unless I'm overcomplicating things).

In Gosu, any draw action can have a mode of :default, :additive or :multiplicative.

**Comment by Mon-Ouie** [comment]
I want to allow per-drawable blending function, yeah (though I'd think just generating a white mask for the image does what you need).

**Comment by bil-bas** [comment]
I have created a "white mask", which I then scale and colour based on what is generating the light. Just using a default blending, however, makes something more white, not more bright.

Alpha Channel - compare
![additive](http://f.cl.ly/items/0Z0m0x3F2y0v272K1o2A/alpha_channel_additive.png)
 to
![regular](http://f.cl.ly/items/0C090E372g1I3k1k402T/alpha_channel_non-additive.png)

As I hope can be seen, "light" in the first one makes the background colourful AND brighter, but the latter one makes it more colourful.
EDIT: Changed the images, since they were from very different versions of the game, so non-comparible.
